src/stores/document-store.js

```diff
@@ -54,6 +54,10 @@ class DocumentStore {
     return this.collection.distinct('key');
   }
 
+  getDocumentsMetadataByKeys(keys, { session } = {}) {
+    return this.collection.find({ key: { $in: keys } }, { session }).toArray();
```

Comment:
There is no index on the `key` in documents, only on `_id` (which has the same value)

Reply:
Funny, did not realize that documents don't have a `key` index. I agree, maybe we can get around to removing this field eventually. Till then I decided to add the index, to make sure we don't suffer from this.

Reply:
:+1: